code/drasil-lang/Language/Drasil/Chunk/Function.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+module Language.Drasil.Chunk.Function (
+  FuncDefn(..),
+  mkFuncDefn, mkFuncDefn', mkFuncDefnByQ
+) where
+
+import Control.Lens ((^.), view, makeLenses)
+
+import Language.Drasil.Chunk.UnitDefn (unitWrapper, IsUnit, MayHaveUnit(..), UnitDefn)
+import Language.Drasil.Classes.Core (HasSymbol(..), HasUID(..))
+import Language.Drasil.Classes (ConceptDomain(..), Display(..), DefiningExpr(..),
+  Quantity, Definition(..), HasSpace(..), Idea(..), NamedIdea(..))
+import Language.Drasil.Chunk.Quantity (mkQuant', QuantityDict)
+import Language.Drasil.Expr.Display (defines)
+import Language.Drasil.Expr (Expr(FCall, C))
+import Language.Drasil.NounPhrase.Core (NP)
+import Language.Drasil.Space (mkPrimitiveMapping, Space)
+import Language.Drasil.Sentence (Sentence(EmptyS))
+import Language.Drasil.UID (UID)
+
+-- TODO: Should we have named arguments for functions? Probably in CodeExpr, but I don't think so in Expr. Perhaps the FCalls need to be different!
+data FuncDefn = FD {
```

Comment:
So: do we really need this? What I mean is, do we need this at the Chunk level? What is different about functions that they need their own 'thing'?  In other words, could we re-use `QDefinition` for functions too?

Reply:
I think we can get away with re-using `QDefinition`s for functions too. Let me try to do this and get back to you.

Reply:
Changed this now.